py-polars/src/polars/functions/lit.py

```diff
@@ -84,7 +83,11 @@ def lit(
     elif isinstance(dtype, type) and issubclass(dtype, BaseExtension):
         msg = f"dtype '{dtype}' is a BaseExtension class, it should be an instance"
         raise TypeError(msg)
-    elif isinstance(value, datetime):
+    elif dtype and dtype.is_object():
```

Comment:
We don't know yet that `dtype` actually is a proper `DataType` so it may not have `is_object`.